### PR TITLE
Feedback rework

### PR DESCRIPTION
**Pull request by amielmon**
Move functionality of feedback header into feedback modal

### CODE REVIEW

File: frontend/components/ResultsPage/FeedbackModal/FeedbackModalInitial.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+
+interface InitialStepProps {
+  setYes: (b:boolean) => void;
+}
+
+export default function FeedbackModalInitial({ setYes } : InitialStepProps) {
```
Comment (NEUDitao) [inline frontend/components/ResultsPage/FeedbackModal/FeedbackModalInitial.tsx:10]: something something purpose statement something something design recipe

File: frontend/components/ResultsPage/FeedbackModal/FeedbackModal.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+  const modalRef = useRef(null);
+  useClickOutside(modalRef, open, setOpen);
+
+  const keyString = 'MODAL'
+  const [show, setFinished] = useFeedbackSchedule(keyString, 86400000);
+
+  const feedbackOptions = ['Class time', 'Professor', 'Prereqs', 'Something else'];
+
+  function handleSubmit() {
+    switch (step) {
```
Comment (NEUDitao) [inline frontend/components/ResultsPage/FeedbackModal/FeedbackModal.tsx:34]: can we make the step initial|checkbox|free an enum?

File: frontend/components/ResultsPage/FeedbackModal/FeedbackModalFree.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+          <input type='radio' id='freeSelectorCenter' name='FeedbackTypeSelector' defaultChecked />
+          <label className='FeedbackModal__freeSelectorCenter' htmlFor='freeSelectorCenter' onClick={ () => setFeedbackType('filter') }>
+            Filter
+          </label>
+          <input type='radio' id='freeSelectorRight' name='FeedbackTypeSelector' />
+          <label className='FeedbackModal__freeSelectorRight' htmlFor='freeSelectorRight' onClick={ () => setFeedbackType('other') }>
+            Other
+          </label>
+        </div>
+        <textarea
```
Comment (NEUDitao) [inline frontend/components/ResultsPage/FeedbackModal/FeedbackModalFree.tsx:36]: not too familiar  with textareas and really tired, but do we need to encode their feedback in some way to make sure we don't get haxored?

**Review by NEUDitao — CHANGES_REQUESTED** [review]

File: frontend/components/ResultsPage/FeedbackModal/FeedbackModalFree.tsx (hunk 3 above)
Comment (amielmon) [reply frontend/components/ResultsPage/FeedbackModal/FeedbackModalFree.tsx:36]: pretty sure react is safe from xss, so this shouldn't be a concern

File: frontend/components/ResultsPage/FeedbackModal/FeedbackModal.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+
+export default function FeedbackModal() {
+  const [open, setOpen] = useState(false);
+  const [yes, setYes] = useState(false);
+  const [selectedFeedback, setSelectedFeedback] = useState([]);
+  const [feedbackQuery, setFeedbackQuery] = useState('');
+  const [feedbackType, setFeedbackType] = useState('filter');
+  const [submitted, setSubmitted] = useState(false);
+  const [step, setStep] = useState('initial');
```
Comment (ryandrew14) [inline frontend/components/ResultsPage/FeedbackModal/FeedbackModal.tsx]: Maybe you want to use `useReducer` here?

File: frontend/components/ResultsPage/useClickOutside.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import { useEffect } from 'react';
+
+// Hook for closing components when clicked outside
+export default function useClickOutside(ref: React.RefObject<HTMLElement>, flag: boolean, setFlag: (b: boolean) => void) {
```
Comment (ryandrew14) [inline frontend/components/ResultsPage/useClickOutside.ts:10]: This is hot

**Review by ryandrew14 — COMMENTED** [review]
Nice hooks

File: frontend/components/ResultsPage/FeedbackModal/FeedbackModal.tsx (hunk 4 above)
Comment (amielmon) [reply frontend/components/ResultsPage/FeedbackModal/FeedbackModal.tsx]: Hmm, I have a lot of state but they're all independent of each other (changing the state of one has no effect on the other pieces of state). Not sure if I need to group all of them into a reducer. What were your thoughts?

**Review by NEUDitao — APPROVED** [review]

**Pull request merged**